lib/include/prjxray/xilinx/bitstream_reader.h

```diff
@@ -98,6 +98,32 @@ BitstreamReader<Series7>::InitWithBytes(T bitstream) {
 	return BitstreamReader<Series7>(std::move(words));
 }
 
+template <>
+template <typename T>
+absl::optional<BitstreamReader<UltraScale>>
+BitstreamReader<UltraScale>::InitWithBytes(T bitstream) {
```

Comment:
`<UltraScale>::InitWithBytes` and `<Series7>::InitWithBytes` should probably just use a template function?  They seem identical except for `s/Series7/UltraScale/`?

Reply:
Yes, `Configuration<UltraScale>::InitWithPackets` along with `Configuration<UltraScale>::createType2ConfigurationPacketData` are other candidates for such a change.